packages/account/src/providers/provider.ts

```diff
@@ -1109,17 +1113,16 @@ Supported fuel-core version: ${supportedVersion}.`
     { signatureCallback }: TransactionCostParams = {}
   ): Promise<Omit<TransactionCost, 'requiredQuantities'>> {
     const txRequestClone = clone(transactionRequestify(transactionRequestLike));
-    const isScriptTransaction = txRequestClone.type === TransactionType.Script;
     const updateMaxFee = txRequestClone.maxFee.eq(0);
 
     // Remove gasLimit to avoid gasLimit when estimating predicates
-    if (isScriptTransaction) {
+    if (isTransactionTypeScript(txRequestClone)) {
       txRequestClone.gasLimit = bn(0);
     }
 
     const signedRequest = clone(txRequestClone) as ScriptTransactionRequest;
     let addedSignatures = 0;
-    if (signatureCallback && isScriptTransaction) {
+    if (signatureCallback && isTransactionTypeScript(signedRequest)) {
```

Comment:
If this was [cast](https://github.com/FuelLabs/fuels-ts/pull/2827/files#diff-69a6670d0426c74935ed2830ce552e7df2379c2d112a582306e57569fcd17029R1123) to a `ScriptTransactionRequest` the latter half of this check should be redundant.

Reply:
Good spot, I actually think the cast is incorrect here.